osu.Game.Tests/Database/RealmLiveTests.cs

```diff
@@ -23,9 +23,9 @@ public void TestLiveEquality()
         {
             RunTestWithRealm((realmFactory, _) =>
             {
-                ILive<BeatmapInfo> beatmap = realmFactory.CreateContext().Write(r => r.Add(new BeatmapInfo(CreateRuleset(), new BeatmapDifficulty(), new BeatmapMetadata()))).ToLive(realmFactory);
+                ILive<BeatmapInfo> beatmap = realmFactory.Run(realm => realm.Write(r => r.Add(new BeatmapInfo(CreateRuleset(), new BeatmapDifficulty(), new BeatmapMetadata()))).ToLive(realmFactory));
```

Comment:
Could have a `T Write<T>` overload in `RealmContextFactory` and shorten this to:
```csharp
ILive<BeatmapInfo> beatmap = realmFactory.Write(r => r.Add(new BeatmapInfo(CreateRuleset(), new BeatmapDifficulty(), new BeatmapMetadata()))).ToLive(realmFactory);
```

Reply:
Considering the only usage of this pattern is in tests, I'll leave it for now. We can add in the future if it becomes a common thing.